### PR TITLE
[wip] .NET Core support

### PR DESCRIPTION
**Pull request by slozier**
Adding .NET Core support. Still working on it. Pull request is for review/comments.

### CODE REVIEW

**Comment by slozier** [comment]
@simplicbe The DLR and the main IronPython DLL compile. I just need someone to look over the changes I've made to make sure they don't change how things behave.

File: Languages/IronPython/IronPython.Modules/array.cs [hunk 1]
```diff
@@ -876,7 +876,7 @@ private class ArrayData<T> : ArrayData {
                     if (!(value is T)) {
                         object newVal;
                         if (!Converter.TryConvert(value, typeof(T), out newVal)) {
-                            if (value != null && typeof(T).IsPrimitive && typeof(T) != typeof(char))
+                            if (value != null && typeof(T).GetTypeInfo().IsPrimitive && typeof(T) != typeof(char))
```
Comment (slide) [inline Languages/IronPython/IronPython.Modules/array.cs]: This doesn't work on v2 or 4.0 builds does it?

File: Languages/IronPython/IronPython.Modules/project.json [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+        "keyFile": "../../../Solutions/DevelKey.snk"
+      }
+    },
+    "Release": {
+      "buildOptions": {
+        "keyFile": "../../../Solutions/Key.snk"
+      }
+    }
+  },
+  "dependencies": {
```
Comment (slide) [inline Languages/IronPython/IronPython.Modules/project.json]: Is there a way to use variables for these?

File: Languages/IronPython/IronPython.Modules/project.json [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+        "Microsoft.Phone": "1.0.0",
+        "Microsoft.Scripting.Core": "1.0.0",
+        "mscorlib": "1.0.0",
+        "mscorlib.Extensions": "1.0.0",
+        "System": "1.0.0",
+        "System.Core": "1.0.0",
+        "System.Net": "1.0.0"
+      }
+    },
+    "sl4": {
```
Comment (slide) [inline Languages/IronPython/IronPython.Modules/project.json]: I think we can remove sl4, we're not packaging it right now at all.

File: Languages/IronPython/IronPython.Modules/array.cs (hunk 1 above)
Comment (slozier) [inline Languages/IronPython/IronPython.Modules/array.cs]: I've got an extension method for platforms which don't support `GetTypeInfo` that just returns the `Type` so it works.

File: Languages/IronPython/IronPython.Modules/project.json (hunk 2 above)
Comment (slozier) [inline Languages/IronPython/IronPython.Modules/project.json]: No idea, I'll look into it...

File: Languages/IronPython/IronPython.Modules/project.json (hunk 3 above)
Comment (slozier) [inline Languages/IronPython/IronPython.Modules/project.json]: :+1:

**Comment by slozier** [comment]
Closing this as it is badly out of date.

**Pull request closed without merging**